### DISCUSSION TITLE
How to write the new version number to pyproject.toml?

### DISCUSSION THREAD

**Discussion by mcnesium** [semantic-release/semantic-release#2188] [Q&A]
I use `semantic-release` to create tags and releases in my instance of GitLab using GitLab CI. Therefore I have my own Docker image which I use with the following pipeline job in `.gitlab-ci.yml`:

    release-semantic:
      image: my-registry/semantic-release:latest
      stage: deploy
      script: semantic-release --plugins "@semantic-release/gitlab" --plugins "@semantic-release/release-notes-generator"

This works fine, but it is missing one thing: how and when will the version number in `pyproject.toml` in the root folder be updated?

There must be some writing to the file plus pushing it back to the repo happening somewhere, right? The semantic-release job has a `GITLAB_TOKEN` able to use the `api` scope, so I figure it should be able to do that. But how should this be done?

**Comment by travi** [comment]
you likely want to use the [exec plugin](https://github.com/semantic-release/exec) to update the file you mentioned and the [git plugin](https://github.com/semantic-release/git) if you need to commit it back to the repository

information about our plugins can be found [in our docs](https://semantic-release.gitbook.io/semantic-release/extending/plugins-list)
